controllers/articles-controllers.js

```diff
@@ -29,14 +31,33 @@ exports.getArticleCommentsById = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  const query = req.query
-  fetchArticles(query)
-    .then((articles) => {
-      res.status(200).send(articles);
-    })
-    .catch((err) => {
-      next(err);
+  const query = req.query;
+  const queryKeys = Object.keys(query);
+  if (queryKeys.length === 0) {
+    fetchArticles(query)
+      .then((articles) => {
+        res.status(200).send(articles);
+      })
+      .catch((err) => {
+        next(err);
+      });
+  } else {
+    fetchTopics().then((topics) => {
+      const topicMatches = topics.filter((topic) => topic.slug === query.topic);
```

Comment:
We don't need to filter here like this. If we make a query to our database with a topic that doesn't exist, our error handling should let us know this rather than filtering out here. You have the right idea with the Promise.reject, but it's in the wrong place I think